packages/aws-rfdk/lib/lambdas/nodejs/lib/custom-resource/simple-resource.ts

```diff
@@ -78,32 +78,32 @@ export abstract class SimpleCustomResource {
     let cfnData: object | undefined;
 
     console.log(`Handling event: ${JSON.stringify(event)}`);
-    const requestType: string = event.RequestType;
     const resourceProperties: object = event.ResourceProperties ?? {};
     const physicalId: string = calculateSha256Hash(resourceProperties);
 
     try {
-      if (requestType === 'Create') {
-        if (!this.validateInput(resourceProperties)) {
-          throw Error(`Input did not pass validation check. Check log group "${context.logGroupName}" ` +
-            `for log stream ${context.logStreamName} for additional information.`);
-        }
-        cfnData = await this.doCreate(physicalId, resourceProperties);
-        console.debug(`Create data: ${JSON.stringify(cfnData)}`);
-      } else if (requestType === 'Update') {
-        if (!this.validateInput(resourceProperties)) {
-          throw Error('Input did not pass validation check');
-        }
-        const oldResourceProperties: object = event.OldResourceProperties ?? {};
-        const oldPhysicalId: string = calculateSha256Hash(oldResourceProperties);
-        if (oldPhysicalId !== physicalId) {
-          console.log('Doing Create -- ResourceProperties differ.');
-          cfnData = await this.doCreate(physicalId, resourceProperties);
-          console.debug(`Update data: ${JSON.stringify(cfnData)}`);
-        }
-      } else {
-        await this.doDelete(physicalId, resourceProperties);
+      const timeout = (prom: any, time: number, exception: any) => {
+        let timer: any;
+        return Promise.race([
+          prom,
+          new Promise((_r, rej) => timer = setTimeout(rej, time, exception)),
+        ]).finally(() => clearTimeout(timer));
+      };
+
+      const defaultReserveTimeMs = 3000;
+      const remainingTimeMs = context.getRemainingTimeInMillis();
+      let reserveTimeMs = Math.min(0.2 * remainingTimeMs, defaultReserveTimeMs);
+      if (reserveTimeMs < defaultReserveTimeMs) {
+        console.debug(`The remaining Lambda execution time of ${reserveTimeMs} ` +
+        `ms might not be sufficient to send a CloudFormation response. At least ${defaultReserveTimeMs} ms is required. ` +
+        'Please increase the Lambda timeout.');
       }
```

Comment:
It's not obvious where the `3000` is coming from or how we chose the values in `0.2 * remainingTimeMs`. Could you add a comment to this section about what we're doing?

Reply:
I've added explanation for the purpose of reserved time in a new commit. The values don't have much background behind them. We simply want to finish earlier than Lambda's timeout, so I first used just 1 second. But we also want to be sure, that we will have enough time to notify CloudFormation about the failure, thus we increased that time to 3 seconds.

The smallest timeout in our code is `30` seconds. So we would keep `27` seconds for the actual function execution. That's why I didn't want to increase the reserved time from `3 seconds` to an even bigger number.